pytest-selenium/tests/test_attribution.py

```diff
@@ -128,9 +129,142 @@ def test_access_free_url_in_citation_text_shows_url_for_default_page(
     attribution.click_attribution_link()
 
     # THEN: The access for free at url in the the citation section should reference the default page of the book
-    default_page_slug = utility.get_default_page(book_slug)
+    default_page_slug = get_default_page(book_slug)
     attribution_access_free_url_expected = (
         "https://openstax.org/books/" + book_slug + "/pages/" + default_page_slug
     )
 
     assert attribution_access_free_url_expected == attribution.access_free_url
+
+
+@markers.test_case("C623700", "C623701")
+@markers.parametrize("page_slug", ["preface"])
+@markers.desktop_only
+@markers.nondestructive
+def test_license_details(selenium, base_url, page_slug):
+    """Verify license name, link address and copyright name"""
+    book_list = Library()
+    book_slugs = book_list.book_slugs_list
+
+    # Repeat the test for all books in the library
+    for book_slug in list(book_slugs):
+
+        # GIVEN: Book page is loaded
+        book = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+        attribution = book.attribution
+
+        # Skip any notification/nudge popups
+        while book.notification_present:
+            book.notification.got_it()
+
+        if book_slug in (
+            "business-law-i-essentials",
+            "calculus-volume-1",
+            "calculus-volume-2",
+            "calculus-volume-3",
+            "principles-financial-accounting",
+            "principles-managerial-accounting",
+        ):
+            license_name_expected = (
+                "Creative Commons Attribution-NonCommercial-ShareAlike License 4.0"
+            )
+            license_url_expected = "http://creativecommons.org/licenses/by-nc-sa/4.0/"
+
+        else:
+            license_name_expected = "Creative Commons Attribution License 4.0"
+            license_url_expected = "http://creativecommons.org/licenses/by/4.0/"
+
+        # WHEN: The attribution section is expanded
+        attribution.click_attribution_link()
+
+        # THEN: License name & license link address is Creative Commons
+        license_name = attribution.citation_builder.get_attribute("text").strip()
+        license_url = attribution.citation_builder.get_attribute("href").strip()
+
+        if license_name == license_name_expected:
+            pass
+        elif license_name == "CC BY":
+            print(
+                f"{book_slug} has license name as {license_name}. But expected name is {license_name_expected}"
+            )
+        else:
+            pytest.xfail(
+                f"{book_slug} has license name as {license_name}. But expected name is {license_name_expected}"
+            )
```

Comment:
Consider reworking by removing the first part of the `if`, changing the `elif` to be:
`if license_name == "CC BY":`
and removing the `else` clause because you're testing this same case in the next `assert`

Reply:
By `because you're testing this same case in the next assert` do you mean lines 195 - 197? It is a different parameter (license_url) that it checks. It fails straight if the assertion fails.

I am not quite sure I understand the removing else clause. 
I want to make sure the license name is equal to the value stored in `license_name_expected` . Thats the ideal scenario.

But some books are hanging on with `cc by`. That is still ok to pass but I need a list to make sure no new books are added to this group with `cc by` till CE team fixes this issue.

The elif condition, checks if any unknown value/placeholder other than the above two appears (negative testing).

Please let me know if I am still not getting what you are trying to explain.

Reply:
Ah...I misread the assert vars.

I would still remove the first `if` clause (as it doesn't do anything), change the `elif` to an `if` and change the final `else` statement to an `elif` so you end up with:
`if license_name == "CC BY":`
and
`elif license_name != license_name_expected:`